File: IUH_Olympic_ICPC/Olympic_Sieucup_2020/B/MXOR.cpp
#include "MXORLIB.h"
#define ll int

main(){
    ll n = get_n();
    assert(n<=17);
    vector<pair<ll, ll>> ans;
    set<ll> stI, stJ;
    for (ll i=1; i<=n; i++) stJ.insert(i);

    for (ll i=1; i<=n; i++)
    {
        stI.insert(i);
        stJ.erase(i);
        vector<ll> I, J;
        for (ll x: stI) I.push_back(x);
        for (ll x: stJ) J.push_back(x);
        ll res=max_xor(I, J);
        ans.push_back({res, -i});
        stI.erase(i);
        stJ.insert(i);
    }
    sort(ans.begin(), ans.end(), greater<pair<ll, ll>>());
    ll x=-ans[0].second;
    ll y=-ans[1].second;
    answer(x, y);
}

File: IUH_Olympic_ICPC/Olympic_Sieucup_2020/B/MXOR.java
//package mxor;
public class MXOR {

    public static void main(String[] argv) {
        int n = MXORLIB.get_n();
        int x = -1, y = -1, best = -1;
        for (int i = 1; i <= n; ++i) {
            for (int j = i + 1; j <= n; ++j) {
                int[] I = new int[1], J = new int[1];
                I[0] = i;
                J[0] = j;
                int tmp = MXORLIB.max_xor(I, J);
                if (tmp > best) {
                    best = tmp;
                    x = i;
                    y = j;
                }
            }
        }
        MXORLIB.answer(x, y);
    }

}
